Assets/Scripts/Dashboard/Button.cs

```diff
@@ -4,24 +4,34 @@
 using UnityEngine.Events;
 using Anivision.Core;
 using Anivision.PlayerInteraction;
+using TMPro;
 
 public class Button : MonoBehaviour
 {
+    [TextArea(3, 10)] public string buttonText;
+    public float buttonCooldownSeconds = 0.5f;
     public Color defaultColor;
     public Color hoverColor;
     [HideInInspector] public UnityEvent onClick;
 
     private InputManager _inputManager;
     private HapticsController _hapticsController;
     private TeleportController _teleportController;
+    private TextMeshPro _TMP;
 
     private void OnEnable()
     {
         _inputManager = InputManager.Instance;
         if (_inputManager == null) throw new System.Exception("Must have an input manager script in the scene");
 
         _hapticsController = HapticsController.Instance;
-        _teleportController = FindObjectOfType(typeof(TeleportController)) as TeleportController;
+        if (_hapticsController == null) throw new System.Exception("Must have a haptics controller in the scene");
+
+        _teleportController = TeleportController.Instance;
+        if (_teleportController == null) throw new System.Exception("Must have a teleport controller in the scene");
```

Comment:
what if you can't teleport in the scene? then a teleport controller isn't necessary right?